File: rb/features.rb
# Opensensemap SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OpensensemapFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpensensemapBaseFeature.new
    when "ratelimit"
      OpensensemapRatelimitFeature.new
    when "retry"
      OpensensemapRetryFeature.new
    when "test"
      OpensensemapTestFeature.new
    when "timeout"
      OpensensemapTimeoutFeature.new
    else
      OpensensemapBaseFeature.new
    end
  end
end

File: py/features.py
# Opensensemap SDK feature factory

from feature.base_feature import OpensensemapBaseFeature
from feature.test_feature import OpensensemapTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpensensemapBaseFeature(),
        "test": lambda: OpensensemapTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
